### ISSUE TITLE
Rename packages to lower case?

### ISSUE THREAD

**Issue by roberth** [NixOS/nixpkgs#188001]
### Describe the problem

- We have only ~65 packages that start with an uppercase letter
- OfBorg is case sensitive when it comes to matching commit prefixes to package names.

### Steps To Reproduce

1. Review a PR for a capital letter package, such as https://github.com/NixOS/nixpkgs/pull/187887
2. OfBorg does not build it automatically (or run the tests)

Alternate steps:

1. Write a package
2. Assume that all package names are lower case. `grpc` is.
3. `SDL` isn't.

### Expected behavior

Consistent naming, automatic CI.

### Additional context

This should be policy or not be done at all.
It should only affect the top level package set. (all-packages)
Other package sets, such as the language-specific ones have their own conventions and requirements.

### Notify maintainers

I think @SuperSandro2000 or @siraben might be interested?

**Comment by SuperSandro2000** [comment]
Yes, agreeing with you that packages should be lower case to make things easier not only for ofborg but also for end users. Only remaining question is if we want to use camelCase or kebab-case if there are capitalized letters in the middle. I would tend to kebab.

**Comment by GabrielDougherty** [comment]
I agree with this proposal and would prefer kebab-case myself. What about package categories, like `python3Packages.*` or `ocaml-ng.ocamlPackages*` - should they be renamed as well - to `python3-packages.*` and `ocaml-ng.ocaml-packages*` respectively?

**Comment by roberth** [comment]
Package sets aren't packages, so I don't think they're inconsistent, and I haven't observed the commit naming issue with them.

**Comment by quantenzitrone** [comment]
some prs i did matching this
- #488486
- #488485
- #488484
- #488483
- #488482
- #488481
- #488480
- #488479
- #488478
- #488477
- #488476
- #488475
- #488474
- #488473
- #488472
- #488471
- #488470
- #488356
- #488353
- #488352
- #488351
- #488349
- #488348
- #488313
- #488305
- #488304
- #488303
- #488302
- #488300
- #488299
- #488298
- #488297
- #488296
- #488295
- #488294
- #488293
- #488292
- #488291
- #488290
- #320363
- #320361
- #320360
- #320317
- #320306
- #320300
- #320299
- #320297
- #320296
- #320295
- #320290
- #320278
- #320275
- #320264
- #320263
- #320260
- #320223

**Comment by quantenzitrone** [comment]
more prs:
- #489642
- #489649
- #489650
- #489651
- #489652
- #489653
- #489654
- #489655
- #489656
- #489657
- #489658
- #489659
- #489660
- #489661
- #489662
- #489663
- #489664
- #489665
- #489666
- #489667
- #489668
